solr/test-framework/src/java/org/apache/solr/SolrTestCaseJ4.java

```diff
@@ -1324,6 +1335,16 @@ public static SolrQueryRequest req(SolrParams params, String... moreParams) {
     return new LocalSolrQueryRequest(h.getCore(), mp);
   }
 
+  /** Generates a SolrQueryRequest randomly assigning multi or single threaded processing. */
+  public static SolrQueryRequest reqRandMulti(SolrParams params, String... moreParams) {
+    ModifiableSolrParams mp = new ModifiableSolrParams(params);
+    for (int i = 0; i < moreParams.length; i += 2) {
+      mp.add(moreParams[i], moreParams[i + 1]);
+    }
+    mp.add("multiThreaded", String.valueOf(random().nextBoolean()));
```

Comment:
If we need to augment parameters, consider SolrParams.wrapDefaults.  It'd be nice if we also had a SolrParams.of(name, value) to make it easy/idiomatic to combine.